### ISSUE TITLE
Godot 4: Callbacks when changing scenes

### ISSUE THREAD

**Issue by jabsatz** [glass-brick/Scene-Manager#11]
**Is your feature request related to a problem? Please describe.**
Would like to solve the confusion/issues proposed in https://github.com/glass-brick/Scene-Manager/issues/8, without internally setting variables on the new scene in an obscure way.
Also would like to give as much control to the users of this library as possible without bloating it.

**Describe the solution you'd like**
A "hook" or "callback" sent to SceneManager when the new scene enters the tree and/or becomes ready. It should recieve the new scene and allow for direct modifications there. I'm imagining something like this:
```
SceneManager.change_scene("res://scene_2.tscn", {
  "on_tree_enter": func(scene): scene.some_variable = "some_value"
})
```
This would be a great feature to implement when Godot 4 releases, since it makes very good use of the new lambda functions feature.

**Additional context**
Unfortunately, I don't think there's a very good way to do this in Godot 3.x.
I don't think a FuncRef would work since the original scene that calls this function would most likely already be unloaded, and the logic in a separate static method seems like it would be more of a hassle than just using a regular Autoload or custom solution.
I also considered using ["expressions"](https://docs.godotengine.org/en/stable/tutorials/scripting/evaluating_expressions.html) and evaluating them internally, but that also feels limiting and extremely hacky.

**Comment by mouring** [comment]
Would be nice to extend this for general purpose as your solution only handles load/reload of a scene, and not fade_in_place().  Right now I have the following hack in my tree:

```
 var default_options := {
 	"speed": 2,
 	"color": Color("#000000"),
[..]
 	"no_scene_change": false,
 	"on_tree_enter": func(scene): null,
 	"on_ready": func(scene): null,
+	"func" : func(): null
 }

[..]
func change_scene(path: Variant, setted_options: Dictionary = {}) -> void:
[..]
 	if not options["no_scene_change"]:
 		_replace_scene(path, options)
+	if options["func"] != null:
+		options["func"].call()
[..]
```
It isn't critical as #12 provides all the functionality I need in the longer form:

```
SceneManager.fade_in_place({ "fade_direction": SceneManager.fade_dir.OUT })
 _FuncCall()
SceneManager.fade_in_place({ "fade_direction": SceneManager.fade_dir.IN })
```

But it would be nice to have the short form of:

```
SceneManager.fade_in_place({ "func": Callable(self, "_FuncCall") })
```

**Comment by jabsatz** [comment]
Yes @mouring , that's definitely doable.
I have no problem adding as many "callbacks" as needed here, since they're a very quick and easy solution to add more options for users.

**Comment by jabsatz** [comment]
This has been implemented and is included from version 1.0.0 🎉 
Closing since Godot 4 IS FINALLY OUT!!!
